### ISSUE TITLE
Skopeo not recognizing auth files 

### ISSUE THREAD

**Issue by simjnd** [containers/skopeo#1198]
Hey guys!
I'm having an issue with Skopeo where if I have an auth file in this format
```json
{
    "auths": {
        "registry": {
            "username": "<value>",
            "password": "<value>"
        }
    }
}
```
and provide it to Skopeo with the --authfile option, I'm getting an error saying `Error initiating layer upload to <repo> in <registry>: unauthorized: The client does not have permission to push to the repository.`. Whereas if, with the same credentials I create an auth file in this format:
```json
{
    "auths": {
        "registry": {
            "auth": "<username:password in base64>"
        }
    }
}
```
Then I can successfully upload. Is it intentional? Most other container tools I've used recognize both formats so it made me think it might be an issue.

Thank you so much for all the work you guys do on Skopeo, this tool is a lifesaver!

**Comment by vrothberg** [comment]
Hi Simon,

Thanks for reaching out!  You conclusion is correct that `username/password` are not supported by Skopeo and other tools using the containers/image library.

> Most other container tools I've used recognize both formats so it made me think it might be an issue.

Can you elaborate which tools support it?  Cross compatibility is usually a good reason to support something.

**Comment by simjnd** [comment]
Both Docker and Kaniko (https://github.com/GoogleContainerTools/kaniko) support this format. Can't remember if Podman does, but iirc it uses the containers/image library so I assume it doesn't?

**Comment by vrothberg** [comment]
> Both Docker and Kaniko (https://github.com/GoogleContainerTools/kaniko) support this format. Can't remember if Podman does, but iirc it uses the containers/image library so I assume it doesn't?

That's correct, Podman also uses c/image. I move this issue over to c/image where we can continue the conversation.

I am entirely open to support the username/password fields but I currently don't have enough time to browse the history whether it's an intentional decision not to.
